cli/src/cast.rs

```diff
@@ -500,6 +500,59 @@ async fn main() -> eyre::Result<()> {
             let selector = contract.abi().functions().last().unwrap().short_signature();
             println!("0x{}", hex::encode(selector));
         }
+        Subcommands::FindBlock { timestamp, rpc_url } => {
+            let ts_target = U256::from(timestamp);
+            let provider = Provider::try_from(rpc_url)?;
+            let last_block_num = provider.get_block_number().await?;
+            let cast_provider = Cast::new(provider);
+            let ts_block = cast_provider.timestamp(last_block_num).await?;
+            let ts_block1 = cast_provider.timestamp(1).await?;
+            let block_num = if ts_block.lt(&ts_target) {
```

Comment:
does that work? I originally tried that but it seemed to not work as expected, which I think is because the timestamps are U256 and therefore comparing their bytes with `<` isn't equivalent to comparing their decimal representation